### ISSUE TITLE
D101 Missing docstring in public class

### ISSUE THREAD

**Issue by peterjc** [biopython/biopython#1961]
We currently enforce various code style issues using ``flake8``, configured via ``Bio/.flake8`` etc and run automatically on TravisCI via tox. More details here including how to install ``flake8`` and the relevant plugins:

https://github.com/biopython/biopython/blob/master/CONTRIBUTING.rst

One of the important docstring checks we currently do not enforce is missing docstrings in public classes.

This issue to to add the remaining missing function docstrings, and then turn on the D101 checks by removing them from the ignore list here:

https://github.com/biopython/biopython/blob/master/Bio/.flake8

Currently there are just 50 of these remaining following work on issue #1203 etc:

```
$ flake8 Bio BioSQL --isolated --select D101 | sort
Bio/Alphabet/__init__.py:173:1: D101 Missing docstring in public class
Bio/Alphabet/__init__.py:211:1: D101 Missing docstring in public class
Bio/Alphabet/__init__.py:236:1: D101 Missing docstring in public class
Bio/Blast/ParseBlastTable.py:16:1: D101 Missing docstring in public class
Bio/Blast/ParseBlastTable.py:32:1: D101 Missing docstring in public class
Bio/Blast/ParseBlastTable.py:54:1: D101 Missing docstring in public class
Bio/CAPS/__init__.py:45:1: D101 Missing docstring in public class
Bio/Cluster/__init__.py:44:1: D101 Missing docstring in public class
Bio/Cluster/__init__.py:48:1: D101 Missing docstring in public class
Bio/Crystal/__init__.py:20:1: D101 Missing docstring in public class
Bio/Entrez/Parser.py:109:1: D101 Missing docstring in public class
Bio/Entrez/Parser.py:118:1: D101 Missing docstring in public class
Bio/Entrez/Parser.py:146:1: D101 Missing docstring in public class
Bio/Entrez/Parser.py:169:1: D101 Missing docstring in public class
Bio/Entrez/Parser.py:187:1: D101 Missing docstring in public class
Bio/Entrez/Parser.py:239:1: D101 Missing docstring in public class
Bio/Entrez/Parser.py:259:1: D101 Missing docstring in public class
Bio/Entrez/Parser.py:281:1: D101 Missing docstring in public class
Bio/Entrez/Parser.py:333:1: D101 Missing docstring in public class
Bio/Entrez/Parser.py:58:1: D101 Missing docstring in public class
Bio/Entrez/Parser.py:68:1: D101 Missing docstring in public class
Bio/Entrez/Parser.py:78:1: D101 Missing docstring in public class
Bio/Entrez/Parser.py:88:1: D101 Missing docstring in public class
Bio/Entrez/Parser.py:98:1: D101 Missing docstring in public class
Bio/ExPASy/ScanProsite.py:75:1: D101 Missing docstring in public class
Bio/ExPASy/ScanProsite.py:95:1: D101 Missing docstring in public class
Bio/FSSP/FSSPTools.py:23:1: D101 Missing docstring in public class
Bio/FSSP/FSSPTools.py:29:1: D101 Missing docstring in public class
Bio/FSSP/__init__.py:142:1: D101 Missing docstring in public class
Bio/FSSP/__init__.py:172:1: D101 Missing docstring in public class
Bio/FSSP/__init__.py:224:1: D101 Missing docstring in public class
Bio/FSSP/__init__.py:45:1: D101 Missing docstring in public class
Bio/FSSP/__init__.py:71:1: D101 Missing docstring in public class
Bio/FSSP/fssp_rec.py:8:1: D101 Missing docstring in public class
Bio/Nexus/Nexus.py:50:1: D101 Missing docstring in public class
Bio/Nexus/Nexus.py:586:1: D101 Missing docstring in public class
Bio/Nexus/Nodes.py:17:1: D101 Missing docstring in public class
Bio/Nexus/Nodes.py:21:1: D101 Missing docstring in public class
Bio/Nexus/StandardData.py:15:1: D101 Missing docstring in public class
Bio/Nexus/Trees.py:28:1: D101 Missing docstring in public class
Bio/PopGen/GenePop/Controller.py:209:1: D101 Missing docstring in public class
Bio/PopGen/GenePop/EasyController.py:16:1: D101 Missing docstring in public class
Bio/SeqIO/_index.py:158:1: D101 Missing docstring in public class
Bio/SeqIO/_index.py:174:1: D101 Missing docstring in public class
Bio/SeqIO/_index.py:41:1: D101 Missing docstring in public class
Bio/SubsMat/FreqTable.py:53:1: D101 Missing docstring in public class
Bio/Wise/psw.py:33:1: D101 Missing docstring in public class
Bio/Wise/psw.py:37:1: D101 Missing docstring in public class
Bio/Wise/psw.py:47:1: D101 Missing docstring in public class
Bio/Wise/psw.py:71:1: D101 Missing docstring in public class
```

This is fairly simple to fix, so I've tagged is as a "good first issue" and would particularly like to encourage new contributors to try working on this (although code authors are of course also welcome to fix their own omissions).

If you want to volunteer and work on a section, please comment here first to avoid duplication of effort. e.g. "I'll work on Bio/Wise/"

If you want to work on this as #1960 together, that's fine too.

**Comment by svalqui** [comment]
I'll work on Bio.Cluste and Bio.CAPS.

**Comment by peterjc** [comment]
Nearly there...

```
$ flake8 Bio BioSQL --isolated --select D101 | sort
Bio/Blast/ParseBlastTable.py:16:1: D101 Missing docstring in public class
Bio/Blast/ParseBlastTable.py:32:1: D101 Missing docstring in public class
Bio/Blast/ParseBlastTable.py:55:1: D101 Missing docstring in public class
```

**Comment by peterjc** [comment]
Once #2072 is merged, we should be able to stop ignoring D101 in the ``.flake8`` file :)

**Comment by MarkusPiotrowski** [comment]
And D102 as well:
```
$ flake8 Bio --isolated --select D102 --statistics
Bio\Blast\NCBIXML.py:536:1: D102 Missing docstring in public method
Bio\Blast\Record.py:98:1: D102 Missing docstring in public method
2     D102 Missing docstring in public method
```

**Comment by peterjc** [comment]
Yes, #2072 will let us close #1963 as well 👍

**Comment by MarkusPiotrowski** [comment]
Finally solved via PR #2072!
